### PR TITLE
Multi target

### PR DESCRIPTION
**Pull request by martijn00**

### CODE REVIEW

**Comment by dansiegel** [comment]
probably worth mentioning here @rotorgames that merging this would simplify a lot of things and eliminate issues with native binaries getting linked out since it would be a single binary for each platform

**Comment by rotorgames** [comment]
Hi, @martijn00. Thank you for your work. 

I tested it. Android and iOS look fine but I can't build the UWP project. It looks like a dependencies issue.
Also, are you sure we need so many props, target files? It looks like we can reuse some of them and clear of duplicates and un usage lines. 
The Appveyor configuration file still uses the old kind of building and publishing. I'm not sure it will work in CI with multi-targeting.

I didn't update projects to multi-targeting before. I can try to fix these issues but it can take much time. It would be awesome if you can help me to resolve that.

Thanks. 

```
Microsoft Visual Studio Enterprise 2019
Version 16.5.1
VisualStudio.16.Release/16.5.1+29920.165
Microsoft .NET Framework
Version 4.8.03752

Xamarin   16.5.000.521 (d16-5@b3ea41e)

Xamarin.Android SDK   10.2.0.100 (d16-5/988c811)
Xamarin.Android Reference Assemblies and MSBuild support.
    Mono: c0c5c78
    Java.Interop: xamarin/java.interop/d16-5@fc18c54
    ProGuard: xamarin/proguard/master@905836d
    SQLite: xamarin/sqlite/3.28.0@46204c4
    Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-5@9f4ed4b

Xamarin.iOS and Xamarin.Mac SDK   13.14.1.39 (30e8706)
Xamarin.iOS and Xamarin.Mac Reference Assemblies and MSBuild support.
```

**Comment by martijn00** [comment]
@rotorgames did you see my messages on Twitter?

File: Directory.build.targets [hunk 1]
```diff
@@ -7,8 +7,8 @@
   </PropertyGroup>
   <PropertyGroup Condition="$(TargetFramework.StartsWith('uap'))">
     <DefineConstants>$(DefineConstants);NETFX_CORE;XAML;WINDOWS;WINDOWS_UWP;UWP</DefineConstants>
-    <TargetPlatformVersion>10.0.16299.0</TargetPlatformVersion>
-    <TargetPlatformMinVersion>10.0.16299.0</TargetPlatformMinVersion>
```
Comment (rotorgames) [inline Directory.build.targets]: Why can't we support 10.0.16299? Xamarin.Forms still support 10.0.16299 and 10.0.14393

Comment (martijn00) [reply Directory.build.targets]: Because Forms advices to support 17763, and will break in the future.

**Pull request merged**